This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: brodieG <brodieG@users.noreply.github.com>
Date: Sat, 12 Nov 2022 13:19:46 +0000
Subject: [PATCH 1/1] MINGW fixes

Originally from Tomas Kalibera AFAIK, possibly
taken from elsewhere.

diff --git a/blosc/blosc-export.h b/blosc/blosc-export.h
index 1111111..2222222 100644
--- a/blosc/blosc-export.h
+++ b/blosc/blosc-export.h
@@ -30,7 +30,7 @@
   #define BLOSC_EXPORT
 #endif  /* defined(BLOSC_SHARED_LIBRARY) */
 
-#if defined(__GNUC__) || defined(__clang__)
+#if !defined(__MINGW32__) && (defined(__GNUC__) || defined(__clang__))
   #define BLOSC_NO_EXPORT __attribute__((visibility("hidden")))
 #else
   #define BLOSC_NO_EXPORT
diff --git a/blosc/blosc.c b/blosc/blosc.c
index 1111111..2222222 100644
--- a/blosc/blosc.c
+++ b/blosc/blosc.c
@@ -57,7 +57,7 @@
 #endif  /* _WIN32 */
 
 /* Include the win32/pthread.h library for all the Windows builds. See #224. */
-#if defined(_WIN32)
+#if 0 && defined(_WIN32)
   #include "win32/pthread.h"
   #include "win32/pthread.c"
 #else
diff --git a/blosc/shuffle.c b/blosc/shuffle.c
index 1111111..2222222 100644
--- a/blosc/shuffle.c
+++ b/blosc/shuffle.c
@@ -14,7 +14,7 @@
 #include "blosc-comp-features.h"
 #include <stdio.h>
 
-#if defined(_WIN32)
+#if !defined(__MINGW32__)
 #include "win32/pthread.h"
 #else
 #include <pthread.h>
